drivers/storage/portworx/component/telemetry.go

```diff
@@ -404,10 +408,43 @@ func (t *telemetry) deleteCCMGoPhonehomeCluster(
 	return nil
 }
 
+// PWX-27401 only reconcile metrics collector if it's already running, revert once collector memory issue got resolved
+func (t *telemetry) shouldReconcileMetricsCollector(
+	cluster *corev1.StorageCluster,
+) error {
+	if t.reconcileMetricsCollector != nil {
+		return nil
+	}
+	// Check if collector V1 or V2 deployment exists
+	existingDeployment := &appsv1.Deployment{}
+	for _, deploymentName := range []string{CollectorDeploymentName, DeploymentNameTelemetryCollectorV2} {
+		err := t.k8sClient.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      deploymentName,
+				Namespace: cluster.Namespace,
+			},
+			existingDeployment,
+		)
+		if err == nil {
```

Comment:
Do we need to check `existingDeployment` or err== nil is enough?

Reply:
if err == nil means the deployment exists, so it should be good.